### PR TITLE
refactor(list): refactor css

### LINKED ISSUE #995: Use a configuration first method (modifiers and states) for List
Relates directly back to #273

### PR DESCRIPTION
**Pull request by christiemolloy**
Closes #995 

Refactor CSS
Modified Handlebars List-Simple-Example to show the nested `<ul>` styles.

### CODE REVIEW

**Comment by mcoker** [comment]
also it seems like this will always present `.pf-c-list` with bullets. And it only allows for spacing of nested `ul`'s, not `ol`'s. can you update this component so that an `ol` has numbers and a `ul` has bullets, and the spacing works for both nested `ul`'s and `ol`'s? Or would you prefer just commit this and we create a separate issue for that work?

**Comment by christiemolloy** [comment]
@mcoker made updates, and created this issue - https://github.com/patternfly/patternfly-next/issues/1086

File: src/patternfly/components/List/list.scss [hunk 1]
```diff
@@ -8,20 +8,20 @@
   --pf-c-list--m-grid--GridGap: var(--pf-global--spacer--md);
 
   // Content variables (no modifier)
+  --pf-c-content--MarginBottom: var(--pf-global--spacer--md);
   --pf-c-content--li--MarginTop: var(--pf-global--spacer--sm);
   --pf-c-content--ul--PaddingLeft: var(--pf-global--spacer--lg);
-  --pf-c-content--ul--MarginTop: var(--pf-global--spacer--md);
-  --pf-c-content--MarginBottom: var(--pf-global--spacer--md);
-  --pf-c-content--ul--MarginLeft: var(--pf-global--spacer--lg);
+  --pf-c-content--ul--MarginTop: var(--pf-c-content--MarginBottom);
```
Comment (mcoker) [inline src/patternfly/components/List/list.scss]: Why would we have `--pf-c-content--MarginBottom`? Couldn't we remove that and this line just be 

```suggestion
  --pf-c-content--ul--MarginTop: var(--pf-global--spacer--md);
```

File: src/patternfly/components/List/list.scss [hunk 2]
```diff
@@ -8,20 +8,20 @@
   --pf-c-list--m-grid--GridGap: var(--pf-global--spacer--md);
 
   // Content variables (no modifier)
+  --pf-c-content--MarginBottom: var(--pf-global--spacer--md);
```
Comment (mcoker) [inline src/patternfly/components/List/list.scss]: I'm curious if you can see any advantage to these vars being namespaced as `--pf-c-content`? shouldn't they just be `--pf-c-list`?

File: src/patternfly/components/List/list.scss [hunk 3]
```diff
@@ -8,20 +8,20 @@
...
   --pf-c-content--ul--PaddingLeft: var(--pf-global--spacer--lg);
-  --pf-c-content--ul--MarginTop: var(--pf-global--spacer--md);
-  --pf-c-content--MarginBottom: var(--pf-global--spacer--md);
-  --pf-c-content--ul--MarginLeft: var(--pf-global--spacer--lg);
+  --pf-c-content--ul--MarginTop: var(--pf-c-content--MarginBottom);
+  --pf-c-content--ul--MarginLeft: var(--pf-c-content--ul--PaddingLeft);
   --pf-c-content--ul--ListStyle: var(--pf-global--ListStyle);
   --pf-c-content--ul--nested--MarginTop: var(--pf-global--spacer--sm);
-  --pf-c-content--ul--nested--MarginLeft: var(--pf-global--spacer--sm);
+  --pf-c-content--ul--nested--MarginLeft: var(---pf-c-content--ul--nested--MarginTop);
```
Comment (mcoker) [inline src/patternfly/components/List/list.scss]: nice! but you have too many `-`'s in `var(---pf-c-content--ul--nested--MarginTop);` :)

**Comment by mcoker** [comment]
closing this in favor of https://github.com/patternfly/patternfly-next/pull/1124

**Pull request closed without merging**